Hardcore.lua

```diff
@@ -471,6 +537,37 @@ function Hardcore:Debug(msg)
 	end
 end
 
+-- Alert UI
+function Hardcore:ShowAlertFrame(style, message)
+	-- style is a string-based key within ALERT_STYLES
+	-- message is any text accepted by FontString:SetText(message)
+
+	message = message or ""
+	
+	local frame, text, icon, file, filename, delay = nil, nil, nil, nil, nil, nil
+	if ALERT_STYLES[style] == nil then
+		Hardcore:Debug("Alert Style not found: " .. style .. ". Using fallback.")
+	end
+	local data = ALERT_STYLES[style] or ALERT_STYLES["hc_red"]
+	frame, text, icon, file, delay = data.frame, data.text, data.icon, data.file, data.delay
+
+	filename = MEDIA_DIR .. file
+	icon:SetTexture(filename)
+	text:SetText(message)
+
+	frame:Show()
+
+	-- TODO: Allow custom sounds per-frame, or allow passing a sound to the function
+	PlaySound(8959)
+
+	-- HACK:
+	-- There's a bug here where a sequence of overlapping notifications share one 'hide' timer
```

Comment:
Possibly assign the ticker to a global and cancel it when a new one opens. Not super important now though